SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+	}
+}
+
+Function GetWSAPasswordPolicyAll {
+	<#
+        .SYNOPSIS
+        Retrieve Workspace One Access servers' Password Policy
+
+        .DESCRIPTION
+        The Get-WSAPasswordPolicyAll cmdlet returns an custom object contains the current configured password policy for Workpace One Access server. 
+		- Validates that network connectivity is possible to Workspace ONE Access
+        - Retrieve Workspace ONE Access Password Policy	
+    #>
+	Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass
+    )
+	
+	$passwordPolicyNodes = New-Object System.Collections.Generic.List[System.Object]
+
+	Try {
+		if (Test-WSAConnection -server $server) {
+			$WSAToken = Request-WSAToken -fqdn $server -user $user -pass $pass
+			if ($WSAToken) {
+				$wsaPasswordPolicy = Get-WSAPasswordPolicy | Select-Object * -ExcludeProperty _links
+				$wsaPasswordLockout = Get-WSAPasswordLockout | Select-Object * -ExcludeProperty _links
+		
+				$wsaTmpObject = @()
+				$wsaTmpObject += [pscustomobject]@{
+				"passwdExpInDays" = [int]$wsaPasswordPolicy.passwordTtlInHours / 24
+				"passwdHistoryRestriction" = $wsaPasswordPolicy.history
+				"passwdMinimumLength" = $wsaPasswordPolicy.minLen
+				"passwdMinUppercase" = $wsaPasswordPolicy.minUpper
+				"passwdMinLowercase" = $wsaPasswordPolicy.minLower
+				"passwdMinNumeric" = $wsaPasswordPolicy.minDigit
+				"passwdMinSpecial" = $wsaPasswordPolicy.minSpecial
+				"passwdMaxConsecutiveIdenticalChar" = $wsaPasswordPolicy.maxConsecutiveIdenticalCharacters
+				"passwdMaxPreviousCharactersReused" = $wsaPasswordPolicy.maxPreviousPasswordCharactersReused
+				"passwdMaxFailAttempts" = $wsaPasswordLockout.numAttempts
+				"passwdAttemptsIntervalInMins" = $wsaPasswordLockout.attemptInterval
+				"passwdUnlockIntervalInMins" = $wsaPasswordLockout.unlockInterval
+				"tempPasswdLifetimeInHour" = $wsaPasswordPolicy.tempPasswordTtl
+				"passwdExpReminderInDay" = [int64]$wsaPasswordPolicy.notificationThreshold / 86400000
+				"passwdExpReminderNotificationFrequencyInDay" = [int64]$wsaPasswordPolicy.notificationInterval / 86400000
+				}
+				
+				$nodeConfigValue = New-Object -TypeName psobject
+				$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $server
+				$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "WSA"
+				$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue ($wsaTmpObject | Select-Object -Skip 0)
+				$passwordPolicyNodes.Add($nodeConfigValue)
+				return $passwordPolicyNodes
+			} else {
+				Write-LogMessage -Type Error -Message "Failed to obtain access token from WSA server '$server',  please verify your credentials and try again"
```

Comment:
Lets use "Workspace ONE Access" in the log messages vs "WSA"

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e